sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -244,10 +246,26 @@ namespace Azure { namespace Core { namespace Test {
     }
 
     // Util for tests getting env vars
-    std::string GetEnv(const std::string& name)
+    std::string GetEnv(std::string const& name)
     {
-      const auto ret = Azure::Core::_internal::Environment::GetVariable(name.c_str());
-
+#if !defined(NDEBUG)
+      // The azure CI pipeline uppercases all EnvVar values from ci.yml files.
+      // That means that any mixed case strings will not be found when run from the CI
+      // pipeline.
+      //
+      // Check to ensure that the filename is upper case
+      {
+        std::string ucName = name;
+        std::transform(ucName.begin(), ucName.end(), ucName.begin(), [](char const& ch) {
+          return static_cast<char>(std::toupper(ch));
+        });
+        if (ucName != name)
+        {
+          throw std::runtime_error("All Azure SDK environment variables must be all upper case.");
+        }
```

Comment:
use `StringExtensions::ToLower` and `StringExtensions::LocaleInvariantCaseInsensitiveEqual` from Azure core

Reply:
Makes sense. I'll make that change.